src/options/pages/blueprints/utils/installableUtils.ts

```diff
@@ -160,6 +120,64 @@ export const getInstalledVersionNumber = (
   return installedExtension?._recipe?.version;
 };
 
+export const isDeployment = (
+  installable: Installable,
+  installedExtensions: UnresolvedExtension[]
+) => {
+  if (isExtension(installable)) {
+    return Boolean(installable._deployment);
+  }
+
+  for (const installedExtension of installedExtensions) {
+    if (
+      installedExtension._recipe?.id === getPackageId(installable) &&
+      installedExtension._deployment
+    ) {
+      return true;
+    }
+  }
+
+  return false;
```

Comment:
This can be improved a little bit.
```suggestion
  const recipeId = installable.metadata.id; // At this point we know that `installable` is a recipe.
  return installedExtensions.some(installedExtestion => installedExtension._recipe?.id === recipeId && recipeExtension?._deployment);
```

Reply:
The condition really depends on the business logic. Is recipe a deployment if at least one of the extensions is a deployment/all extensions are deployments?

In any case, getting the recipe id can be pulled from the loop.

Reply:
Agree it's better to use `some` here. 

It's probably nicer to hoist out the `getPackageId(installable)`. Unlikely it'd make a performance difference either way